### ISSUE TITLE
CompletionSuggester not working because prefix is placed at wrong position in JSON

### ISSUE THREAD

**Issue by Lenoirio** [elastic/elasticsearch-java#150]
I'm migrating from the old high-level-client and I'm having an issue with the CompletionSuggester   (using 7.16.3 right now)

As the doc at https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters.html states, a request looks like this

```json5
POST music/_search?pretty
{
  "suggest": {
    "song-suggest": {
      "prefix": "nir",        
      "completion": {         
          "field": "suggest"  
      }
    }
  }
}
```

As you can see the prefix is placed in front of the "completion"-object. When I use your CompletionSuggester it will place the prefix inside the "completion"-object.  As a result ElasticSearch gives me this error:

[2022-02-01T14:26:53,829][ERROR][c.a.e.AuthenticationPlugin] [SAG-21X1703] org.elasticsearch.xcontent.XContentParseException: [1:257] [completion] unknown field [prefix]

Am I doing something wrong or is this a bug?  The old java-client was doing it right and also placed the prefix in front of the "completion"-object. 

Thank you very much!

**Comment by sothawo** [comment]
That just bit me as well.

The old RHLC code passed the prefix argument to it's parent, so it ended up at the right place:

```java
    public CompletionSuggestionBuilder prefix(String prefix, Fuzziness fuzziness) {
        super.prefix(prefix);
        this.fuzzyOptions = new FuzzyOptions.Builder().setFuzziness(fuzziness).build();
        return this;
    }
```

**Comment by KamilGlo** [comment]
This also occurs for term suggester.
Using a request for suggestion like
```
final SearchRequest request =
    new SearchRequest.Builder()
                     .index(getIndexName())
                     .suggest(s ->
                              s.suggesters(fieldName + "_suggestion",
                                           FieldSuggester.of(f -> f.term(term -> term.field(fieldName)
                                                                                     .text(searchFieldText)
                                                                                     .minWordLength(2)))))
                     .build();
```
builds a request
```
{
  "suggest":
  {
    "title_suggestion":
    {
      "term":
      {
        "field":"title",
        "min_word_length":2,
        "text":"ths is nt rght"
      }
    }
  }
}
```
and server returns
```
co.elastic.clients.elasticsearch._types.ElasticsearchException: [es/search] failed: [parsing_exception] suggester[term] doesn't support field [text]
```

**Comment by swallez** [comment]
This has been fixed in #0a47bd2 and will be part of version 8.2.1 (it's already available in the nightly snapshots).
